### PR TITLE
Associated groups token

### PR DESCRIPTION
**Pull request by jensotto**
See PR #2174

### CODE REVIEW

**Comment by jensotto** [comment]
@jansenbe

File: Core/OfficeDevPnP.Core/Framework/Provisioning/ObjectHandlers/ObjectSiteSecurity.cs [hunk 1]
```diff
@@ -843,6 +849,44 @@ where user.IsSiteAdmin
...
+                    throw new ArgumentOutOfRangeException(nameof(groupType));
+            }
+
+            string createdAssociateGroupsKey = "vti_createdassociategroups";
+            string associatedGroupKey = $"vti_associate{groupTypeText}group";
+
+            if (web.AllProperties.FieldValues.ContainsKey(createdAssociateGroupsKey) &&
+                web.AllProperties.FieldValues.ContainsKey(associatedGroupKey))
+            {
+                string associatedOwnerGroupId = (string)web.AllProperties.FieldValues[associatedGroupKey];
```
Comment (heinrich-ulbricht) [inline Core/OfficeDevPnP.Core/Framework/Provisioning/ObjectHandlers/ObjectSiteSecurity.cs]: `associatedOwnerGroupId` should be more generic, like `associatedGroupId`

File: Core/OfficeDevPnP.Core.Tests/Framework/ObjectHandlers/ObjectSiteSecurityTests.cs [hunk 2]
```diff
@@ -281,6 +281,40 @@ public void CanCreateEntities2()
...
+                creationInfo.IncludeSiteGroups = true;
+                creationInfo.IncludeDefaultAssociatedGroups = true;
+                var template = new ProvisioningTemplate();
+                template = new ObjectSiteSecurity().ExtractObjects(web, template, creationInfo);
+
+                ctx.Load(web,
+                    w => w.AssociatedOwnerGroup.Title,
+                    w => w.AssociatedMemberGroup.Title,
+                    w => w.AssociatedVisitorGroup.Title);
+                ctx.ExecuteQuery();
```
Comment (heinrich-ulbricht) [inline Core/OfficeDevPnP.Core.Tests/Framework/ObjectHandlers/ObjectSiteSecurityTests.cs]: ExecuteQueryRetry?

**Comment by heinrich-ulbricht** [comment]
It got rather quiet since our burst of activity. Since the conference time is over I think we can ramp up activity again :) This pull request is rather important I think. @jansenbe Do you see any chance of reviewing and merging this (replacing my implementation)?

**Comment by heinrich-ulbricht** [comment]
@jansenbe This longer we wait the more complicated it gets. What do you suggest? The changes by @jensotto seemed pretty good.

**Comment by jansenbe** [comment]
We're working on API updates that allow us to detect when the "async" part of the provisioning flow is done...with that in place the current code should be more reliable. Let's re-evaluate this problem once we're at that stage. So for now I'll close this PR, we can for sure re-open it afterward if needed. Thanks for your understanding.

**Pull request closed without merging**